staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -442,6 +448,16 @@ func NewCacherFromConfig(config Config) (*Cacher, error) {
 			}, time.Second, stopCh,
 		)
 	}()
+	if utilfeature.DefaultFeatureGate.Enabled(features.ConsistentListFromCache) {
+		ctx := context.Background()
+		supports, err := cacher.Supports(ctx, storage.RequestWatchProgress)
```

Comment:
I'm worried about one edge case. This code called during apiserver bootstrap for each K8s resource. There can be many resources, so the supports function will be called multiple times. If etcd is down during bootstrap, every call can fail and and take a long time. Even if we set a timeout it will not solve the problem of errors not being cached. With 10 resources, 3 node etcd cluster and 1 second timeout it will add even 30 seconds to apiserver bootstrap.

There are two directions we can go:
* Make the `Supports` logic smarter to avoid calling same endpoint multiple times after getting an error. A behavior that might not be desired when calling `Supports` when serving LIST request
* [preferred] Drop the check to `Supports` from bootstrap

I would recommend to drop this call from bootstrap at all. With the latest version of `DefaultFeatureGate` the feature will be assumed to be disabled by default, removing the need to prefill the cache in `DefaultFeatureGate`.

Reply:
> I would recommend to drop this call from bootstrap at all. With the latest version of DefaultFeatureGate the feature will be assumed to be disabled by default, removing the need to prefill the cache in DefaultFeatureGate.

@serathius - can you clarify this one? Are you saying that `ConsistentListFromCache` feature will be disabled by default even in Beta? How we're going to use it then?

Reply:
> @serathius - can you clarify this one? Are you saying that ConsistentListFromCache feature will be disabled by default even in Beta? How we're going to use it then?

Sorry for confusion. I meant that `DefaultFeatureGate` will be disabled by default, not `ConsistentListFromCache`. I mean that feature will be disabled until we are able to reach etcd and confirm that it supports `storage.RequestWatchProgress`.